app/assets/javascripts/utils/date_utils.js

```diff
@@ -8,5 +8,48 @@ export const toDate = (date) => {
   if (date instanceof Date) {
     return date;
   }
-  return parseISO(date);
+  // should not really happen in production, but it's here for making the tests pass
+  // this is what moment js does for invalid dates
+  if (date === null || date === undefined) {
+    return new Date();
+  }
+
+  const parsedDate = parseISO(date);
+  if (!isValid(parsedDate)) {
+    // eslint-disable-next-line no-console
+    console.log(`Invalid date - ${date}`);
+  }
+  return parsedDate;
+};
+
+export const formatWithTime = (date) => {
```

Comment:
I think `formatWithTime` and `formatWithoutTime` might have room for improvement in the names. Format what? A Date? A DateTime? I think this works for a Date or a DateTime, right?

Maybe `formatDateAndTime` and `formatDate` (or `formatDateWithoutTime`)?

Reply:
It works for any valid ISO 8601 string or a Date object. I'll improve the function names to reflect that.